### DISCUSSION TITLE
Data Importer issue with certain banks only

### DISCUSSION THREAD

**Discussion by noorchestra** [firefly-iii/firefly-iii#9129] [Firefly III]
Hi,

so I've setup _Firefly III_ (6.1.19) and the _Data Importer_ (1.5.3) on Synology (Docker compose installation), and both seem to be working correctly -

I've managed to get data for two banks - _N26_, _Revolut_ (Country chosen _Slovenia_) via GoCardless into Firefly,
but I get errors when trying to add _NovaKBM_ or _Delavska Hranilnica_ (Country chosen _Slovenia_ in this case as well).

I am able to go through the adding process for both banks directly through the test integrations in the _Connect my bank_ section in the GoCardless account, it's only through the Data Importer flow that this fails.

Error received:

```
Whoops! 500 :(

Sorry, the Firefly III Data Importer broke down.
Error message

AuthenticatedJsonPost: Client error: `POST https://bankaccountdata.gocardless.com/api/v2/agreements/enduser/` resulted in a `400 Bad Request` response: {"access_valid_for_days":{"summary":"Incorrect access_valid_for_days","detail":"access_valid_for_days must be > 0 and <= (truncated...)
More information

The error occurred in /var/www/html/app/Services/Nordigen/Request/Request.php:245. 
```

Debug page:

Debug information generated at 2024-08-09 15:57:42 Europe/Ljubljana for Firefly III Data Importer version **1.5.3**.

<table><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>Version</td><td>1.5.3</td></tr><tr><td>Build</td><td><span>#</span>643, base <span>#</span>86</td></tr><tr><td>System</td><td>PHP 8.3.9, Linux, apache2handler</td></tr></table><table><tr><th colspan="2">App information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>Timezone</td><td>Europe/Ljubljana, Europe/Ljubljana</td></tr><tr><td>Environment</td><td>local</td></tr><tr><td>Debug mode</td><td>false, cache 'file'</td></tr><tr><td>Log level</td><td>debug, stack</td></tr><tr><td>Display errors</td><td>Off, ALL errors</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Trusted proxies</td><td>192.x.x.x</td></tr></table><table><tr><th colspan="2">User information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0</td></tr></table>

I randomly tried all banks available under Slovenia in the dropdown, and this is the result:
1. Banks that work (ie. redirected in a new tab to connect with GoCardless): Airwallex, bunq, HSBCnet, Intesa Sanpaolo, Lunar, Monese, N26 Bank, Neteller, NLB Banka, Paypal, Paysera, Revolut, Skrill, Soldo, Stripe, Unicredit Banka Slovenija, Wise
2. Banks that don't work (ie. I got the error above): Addiko Bank, BKS Bank AG, Delavska Hranilnica, Deželna Banka Slovenije, Gorenjska Banka, LON, Nova KBM, Primorska Hranilnica, SKB Banka, Sparkasse

Coincidentally, all the banks that don't work are banks that have the following comment added:
"Only once account can be connected per one connection, for multiple accounts multiple requisitions should be created"
in [this spreadsheet](https://docs.google.com/spreadsheets/d/1EZ5n7QDGaRIot5M86dwqd5UFSGEDTeTRzEq3D9uEDkM/edit?gid=976380583#gid=976380583) provided by GoCardless.

Not sure if this is what's causing this, or what requisitions are and what to do with this.

Hopefully someone can help, I'm really enjoying Firefly III and would really like to get this data in automatically.

Thank you!

**Comment by JC5** [comment]
What number did you fill in during the connection process? Or did you leave it at the default?

**Reply by noorchestra** [reply]
Number? In the Client ID field? 
Same one in all the attempts, working or not (=3), found on the OAuth page after creating the OAuth client.

**Reply by JC5** [reply]
The number of days when you connect a bank :)

**Reply by noorchestra** [reply]
Ah, sorry :) Yes, left it at the default 90. 
I tried to change to 89 in a couple of the attempts as I saw some past issues with this, but got the same error.

**Reply by JC5** [reply]
Try 30 or something.

**Reply by noorchestra** [reply]
Tried 30, 15, 1 with both banks I need -
I got the same error sadly.

**Reply by JC5** [reply]
That looks like a bug, I'll pick it up!
